code/game/objects/items/mail.dm

```diff
@@ -332,3 +352,172 @@
 /obj/item/paper/fluff/junkmail_generic/Initialize(mapload)
 	default_raw_text = pick(GLOB.junkmail_messages)
 	return ..()
+
+/obj/item/mail/traitor
+	var/armed = FALSE
+	var/datum/weakref/made_by
+	// If somehow mind will disappear, admins will still have info about creator
+	var/made_by_cached_name
+	var/made_by_cached_ckey
+	goodie_count = 0
+
+/obj/item/mail/traitor/envelope
+	name = "envelope"
+	icon_state = "mail_large"
+	stamp_max = 2
+	stamp_offset_y = 5
+
+/obj/item/mail/traitor/after_unwrap(mob/user)
+	user.temporarilyRemoveItemFromInventory(src, force = TRUE)
+	playsound(loc, 'sound/items/poster_ripped.ogg', vol = 50, vary = TRUE)
+	for(var/obj/item/stuff in contents) // Mail and envelope actually can have more than 1 item.
+		if(user.put_in_hands(stuff) && armed)
+			log_bomber(user, "opened armed mail made by [made_by_cached_name] ([made_by_cached_ckey]), activating", stuff)
+			INVOKE_ASYNC(stuff, TYPE_PROC_REF(/obj/item/mail/traitor, attack_self), user)
+	qdel(src)
+
+/obj/item/mail/traitor/multitool_act(mob/living/user, obj/item/tool)
+	if(armed == FALSE || user.get_inactive_held_item() != src)
+		. = ..()
+	else
+		if(user.mind == made_by.resolve())
+			balloon_alert(user, "disarming trap...")
+			if(!do_after(user, 2 SECONDS, target = src))
+				return FALSE
+			balloon_alert(user, "disarmed")
+			playsound(src, 'sound/machines/defib_ready.ogg', vol = 100, vary = TRUE)
+			armed = FALSE
+			return TRUE
+		else
+			balloon_alert(user, "tinkering with something...")
+			
+			if(!do_after(user, 2 SECONDS, target = src))
+				after_unwrap(user)
+				return FALSE
+			if(prob(50))
+				balloon_alert(user, "disarmed something...?")
+				playsound(src, 'sound/machines/defib_ready.ogg', vol = 100, vary = TRUE)
+				armed = FALSE
+				return TRUE
+			else
+				after_unwrap(user)
+				return TRUE
+
+/obj/item/storage/mail_counterfeit_device
+	name = "GLA-2 mail counterfeit device"
+	desc = "Device that actually able to counterfeit NT's mail. This device also able to place a trap inside of mail for malicious actions. Trap will \"activate\" any item inside of mail. Also it might be used for contraband purposes. Integrated micro-computer will give you great configuration optionality for your needs."
+	w_class = WEIGHT_CLASS_NORMAL
+	icon = 'icons/obj/device_syndie.dmi'
+	icon_state = "mail_counterfeit_device"
+
+/obj/item/storage/mail_counterfeit_device/Initialize(mapload)
+	. = ..()
+	atom_storage.max_slots = 1
+	atom_storage.allow_big_nesting = TRUE
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+
+/obj/item/storage/mail_counterfeit_device/examine_more(mob/user)
+	. = ..()
+	var/list/msg = list()
+	msg += "[span_notice("<i>You notice the manufacture marking on the side of the device...</i>")]"
```

Comment:
Remove `msg`, add these onto `.`, then `return .` at the end.

You are calling `..()` but ignoring its value, which doesn't look right here.

Reply:
Done